utils/municipality.ts

```diff
@@ -0,0 +1,339 @@
+import { BlobService } from "azure-storage";
+import { array } from "fp-ts/lib/Array";
+import { left as leftE, right as rightE } from "fp-ts/lib/Either";
+import { Option } from "fp-ts/lib/Option";
+import {
+  fromEither,
+  fromLeft,
+  TaskEither,
+  taskEither
+} from "fp-ts/lib/TaskEither";
+import { getRequiredStringEnv } from "io-functions-commons/dist/src/utils/env";
+import * as t from "io-ts";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import { NonEmptyString, PatternString } from "italia-ts-commons/lib/strings";
+import { Municipality } from "../generated/definitions/Municipality";
+import {
+  getCsvFromURL,
+  getFileFromBlob,
+  parseCsv,
+  StringMatrix,
+  writeBlobFromJson
+} from "./file";
+
+const ITALIAN_MUNICIPALITIES_URL = getRequiredStringEnv(
+  "ITALIAN_MUNICIPALITIES_URL"
+);
+
+const MUNICIPALITIES_CONTAINER_NAME = getRequiredStringEnv(
+  "MUNICIPALITIES_CONTAINER_NAME"
+);
+const ABOLISHED_MUNICIPALITIES_BLOB_NAME = getRequiredStringEnv(
+  "ABOLISHED_MUNICIPALITIES_BLOB_NAME"
+);
+const MUNICIPALITIES_CATASTALI_BLOB_NAME = getRequiredStringEnv(
+  "MUNICIPALITIES_CATASTALI_BLOB_NAME"
+);
+const FOREIGN_COUNTRIES_BLOB_NAME = getRequiredStringEnv(
+  "FOREIGN_COUNTRIES_BLOB_NAME"
+);
+
+/**
+ * This type represent the data of :ABOLISHED_MUNICIPALITIES_FILEPATH:
+ */
+export const AbolishedMunicipality = t.type({
+  comune: t.string,
+  istat: t.string,
+  provincia: t.string
+});
+
+export const AbolishedMunicipalityArray = t.readonlyArray(
+  AbolishedMunicipality
+);
+
+export type AbolishedMunicipalityArray = t.TypeOf<
+  typeof AbolishedMunicipalityArray
+>;
+
+export type AbolishedMunicipality = t.TypeOf<typeof AbolishedMunicipality>;
+
+const optionMunicipalitiesWithCatastale = {
+  delimiter: ",",
+  from_line: 1,
+  skip_empty_lines: true,
+  skip_lines_with_error: true,
+  trim: true
+};
+
+const optionCsvParseForeignCountries = {
+  delimiter: ";",
+  from_line: 3, // skip header + italy entry
+  skip_empty_lines: false,
+  skip_lines_with_error: false,
+  trim: true
+};
+
+const currentMiunicipalitiesParserOption = {
+  delimiter: ";",
+  from_line: 4,
+  skip_empty_lines: true,
+  skip_lines_with_error: true,
+  trim: true
+};
+
+export interface ISerializableMunicipality {
+  codiceCatastale: string;
+  municipality: Municipality;
+}
+
+// try to decode municipality csv row in a Municipality object
+export const decodeMunicipality = (
+  record: ReadonlyArray<string>
+): t.Validation<Municipality> => {
+  if (record.length < 13) {
+    return leftE([
+      {
+        context: [],
+        value: "record has not the right length"
+      }
+    ]);
+  }
```

Comment:
I think that this check is trying to avoid column shifting problems, because if a record length is larger than expected, we can assign a wrong value to a property. Moreover we use a positional notation to extract values from csv files.

Reply:
> if a record length is larger than expected

Actually the check is for records **shorter** than expected, isn't it?

Reply:
Yes, but the logic is the same, maybe we can turn this check into `record.length !== <expected_record_length>`